hal/network/lwip/cellular/pppncpnetif.cpp

```diff
@@ -150,41 +165,64 @@ void PppNcpNetif::loop(void* arg) {
 }
 
 int PppNcpNetif::up() {
-    NetifEvent ev = NetifEvent::Up;
-    int r = os_queue_put(queue_, &ev, 0, nullptr);
-    if (r) {
-        // FIXME: we'll miss this event, just as an attempt to potentially recover
-        // set internal state.
-        up_ = true;
+    if (lastNetifEvent_ == NetifEvent::Up) {
+        return SYSTEM_ERROR_NONE;
     }
-    return r;
+    // FIXME:
+    // The following separate sequential atomic operations do not make the whole change atomic.
+    // We may end up in an inconsistent state. Same to wherever the changes being made.
+    lastNetifEvent_ = NetifEvent::Up;
```

Comment:
@XuGuohui Let's at least make sure that these operations are completely atomic:
```c++
if (lastNetifEvent_.exchange(NetifEvent::Up, std::memory_order_acq_rel) == NetifEvent::Up) {
    return SYSTEM_ERROR_NONE;
}
```

Reply:
This comment is actually for line 168?

Reply:
For 168 to 174. Two operations combined into 1 atomic read-modify-write.

Reply:
And also all the other similar cases.

Reply:
Done